src/platform/silabs/wifi/rs911x/WifiInterfaceImpl.cpp

```diff
@@ -70,168 +70,94 @@ osThreadAttr_t kDrvTaskAttr = { .name       = "drv_rsi",
-#if (CHIP_DEVICE_CONFIG_ENABLE_IPV4)
-bool hasNotifiedIPV4 = false;
-#endif /* CHIP_DEVICE_CONFIG_ENABLE_IPV4 */
-bool hasNotifiedWifiConnectivity = false;
-
 #if (RSI_BLE_ENABLE)
 extern rsi_semaphore_handle_t sl_rs_ble_init_sem;
 #endif
 
-// DHCP Poll timer
-static osTimerId_t sDHCPTimer;
 static osMessageQueueId_t sWifiEventQueue = NULL;
 /*
  * This file implements the interface to the RSI SAPIs
  */
 static uint8_t wfx_rsi_drv_buf[WFX_RSI_BUF_SZ];
 static wfx_wifi_scan_ext_t temp_reset;
 
-/******************************************************************
- * @fn   void rsi_wireless_driver_task_wrapper(void * argument)
- * @brief
- *       wrapper thread for the driver task
- * @param[in] argument: argument
- * @return
- *       None
- *********************************************************************/
-static void rsi_wireless_driver_task_wrapper(void * argument)
+sl_status_t TriggerPlatformWifiDisconnection()
 {
-    rsi_wireless_driver_task();
+    VerifyOrReturnError(rsi_wlan_disconnect() == RSI_SUCCESS, SL_STATUS_FAIL);
+    return SL_STATUS_OK;
 }
 
-static void DHCPTimerEventHandler(void * arg)
+CHIP_ERROR GetAccessPointInfo(wfx_wifi_scan_result_t & info)
 {
-    WifiPlatformEvent event = WifiPlatformEvent::kStationDhcpPoll;
-    sl_matter_wifi_post_event(event);
-}
+    int32_t status = RSI_SUCCESS;
+    uint8_t rssi   = 0;
 
-static void CancelDHCPTimer(void)
-{
-    osStatus_t status;
+    info.security = wfx_rsi.sec.security;
+    info.chan     = wfx_rsi.ap_chan;
+    memcpy(&(info.bssid[0]), wfx_rsi.ap_mac.data(), kWifiMacAddressLength);
 
-    // Check if timer started
-    if (!osTimerIsRunning(sDHCPTimer))
-    {
-        ChipLogError(DeviceLayer, "CancelDHCPTimer: timer not running");
-        return;
-    }
+    status = rsi_wlan_get(RSI_RSSI, &rssi, sizeof(rssi));
+    VerifyOrReturnError(status = RSI_SUCCESS, CHIP_ERROR_INTERNAL);
 
-    status = osTimerStop(sDHCPTimer);
-    if (status != osOK)
-    {
-        ChipLogError(DeviceLayer, "CancelDHCPTimer: failed to stop timer with status: %d", status);
-    }
+    info.rssi = (-1) * rssi;
+
+    return CHIP_NO_ERROR;
 }
 
-static void StartDHCPTimer(uint32_t timeout)
+CHIP_ERROR GetAccessPointExtendedInfo(wfx_wifi_scan_ext_t & info)
 {
-    osStatus_t status;
+    uint8_t buff[RSI_RESPONSE_MAX_SIZE] = { 0 };
 
-    // Cancel timer if already started
-    CancelDHCPTimer();
+    int32_t status = rsi_wlan_get(RSI_WLAN_EXT_STATS, buff, sizeof(buff));
+    VerifyOrReturnError(status == RSI_SUCCESS, CHIP_ERROR_INTERNAL,
+                        ChipLogError(DeviceLayer, "Failed, Error Code : 0x%lX", status));
 
-    status = osTimerStart(sDHCPTimer, pdMS_TO_TICKS(timeout));
-    if (status != osOK)
-    {
-        ChipLogError(DeviceLayer, "StartDHCPTimer: failed to start timer with status: %d", status);
-    }
-}
+    // TODO: We need to clean up this casting for the extended information
+    rsi_wlan_ext_stats_t * test = (rsi_wlan_ext_stats_t *) buff;
```

Comment:
imo this should all be the other way around
buff (or a new name) should be of type `rsi_wlan_ext_stats_t`. and casted a a `uint8_t* `for `rsi_wlan_get` 
Ideally `rsi_wlan_get` API gets changed to expect a rsi_wlan_ext_stats_t pointer or reference type.

Reply:
done